File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/X. Convert To Decimal 2/main.cpp
// https://codeforces.com/group/MWSDmqGsZm/contest/326175/problem/X
#include <iostream>
#include <cstdio>

using namespace std;

int main()
{
    unsigned short t;
    cin >> t;
    while (t-- > 0)
    {
        unsigned int n;
        cin >> n;
        unsigned int ones = 0;
        while (n > 0)
        {
            ones += 1 & n;
            n >>= 1;
        }
        unsigned int bNum = 1;
        while (ones-- > 1)
        {
            bNum <<= 1;
            bNum ^= 1;
        }

        cout << bNum << endl;
    }
}

File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/X. Convert To Decimal 2/main.cs
// https://codeforces.com/group/MWSDmqGsZm/contest/326175/problem/X

using System;
using static System.Console;

namespace Codeforces.Assuit.Contest1
{
    class ProblemX
    {
        static void Main(string[] args)
        {
            ushort t = ushort.Parse(Console.ReadLine());
            while (t-- > 0)
            {
                uint n = uint.Parse(Console.ReadLine());
                uint ones = 0;
                while (n > 0)
                {
                    ones += 1 & n;
                    n >>= 1;
                }
                uint bNum = 1;
                while (ones-- > 1)
                {
                    bNum <<= 1;
                    bNum ^= 1;
                }
                WriteLine(bNum);
            }
        }
    }
}
